services/QuillLMS/spec/models/unit_activity_spec.rb

```diff
@@ -146,19 +142,66 @@
 
   context 'self.get_classroom_user_profile' do
     it 'get user profile data' do
-      units = UnitActivity.get_classroom_user_profile(classroom.id, student.id)
-      expect(units.count).to eq(2)
+      unit_activities = UnitActivity.get_classroom_user_profile(classroom.id, student.id)
+      expect(unit_activities.count).to eq(2)
     end
 
     it 'gracefully account for nils' do
-      units = UnitActivity.get_classroom_user_profile(nil, nil)
-      expect(units.count).to eq(0)
+      unit_activities = UnitActivity.get_classroom_user_profile(nil, nil)
+      expect(unit_activities.count).to eq(0)
     end
 
-    it 'includes units even if their activities are archived' do
+    it 'includes unit_activities even if their activities are archived' do
       activity.update(flags: [Activity::ARCHIVED])
-      units = UnitActivity.get_classroom_user_profile(classroom.id, student.id)
-      expect(units.count).to eq(2)
+      unit_activities = UnitActivity.get_classroom_user_profile(classroom.id, student.id)
+      expect(unit_activities.count).to eq(2)
     end
+
+    describe 'publish dates' do
+
+      it 'includes unit activities that do not have a publish date' do
+        unit_activity.update(publish_date: nil)
+        lessons_unit_activity.update(publish_date: nil)
+        unit_activities = UnitActivity.get_classroom_user_profile(classroom.id, student.id)
+        expect(unit_activities.count).to eq(2)
+      end
+
+      it 'includes unit activities that have a publish date that have already passed in the timezone of the teacher' do
```

Comment:
Per my note on timezones in the query, I don't think you actually need to do anything related to timezones in the specs for the query.  The one thing we do need to do is make sure that if the teacher does have a timezone set, that we apply said timezone when we write the `publish_date` to the database, but I think that's actually going to be in a different PR since there's no code in this one to handle that writing.

Reply:
@anathomical That's an interesting point, and one I hadn't really considered (saving the publish date itself in the teacher's timezone, instead of in UTC). Newer versions of postgresql do have a data type for date times with time zones, but we'd have to do a funky migration to store it that way (by default the `datetime` columns are saved as `timestamp without time zone` data types). I think this is the first and only place in our database we'd be introducing that. I think it's likely, but I'm not sure, that the new data type would translate cleanly into our frontend without having to make additional adjustments. Saving the timezone on the publish date itself also means that adjusting the set time zone wouldn't automatically adjust the publish dates, which may or may not be behavior we want. Due dates are not currently saved with timezones, though we could update that. I'm not sure whether the simplification of this query is worth introducing the new data type or not - what do you think?

Reply:
@emilia-friedberg Sorry, I think I was unclear.  What I meant was that we need to make sure that whatever payload we pass from the front-end with datetime data in it includes a timezone so that PostgreSQL can accurately know how to convert that value to UTC for storage in the database.  I think that you are totally correct that we shouldn't store timezone offsets directly on the column, even if PostgreSQL makes that possible as that would be a big change in how we handle all other datetimes.

Shorter version: I don't think we need timezone-related specs on this PR since we're just going to rely on PostgreSQL to manage UTC conversion.  We should have some timezone-related specs on whatever code we write that sets `publish_date`, but that's going to be in a different PR as I understand it.